pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -427,6 +432,29 @@ func (p *ConfigurationEditFlags) Apply(
 		servinglib.UpdateUser(template, p.PodSpecFlags.User)
 	}
 
+	if cmd.Flags().Changed("scale-init") {
+		containsAnnotation := func(annotationList []string, annotation string) bool {
+			for _, element := range annotationList {
+				if strings.Contains(element, annotation) {
+					return true
```

Comment:
we should do an exact match for the respective annotation key here, because we support unsetting an annotation by appending `-` to anno key, and that will match the case here.

Reply:
Well, in either case (set/unset), if user specifies both way, we should prompt that, so we can keep this check as is.